_pytest/capture.py

```diff
@@ -252,6 +252,8 @@ def writelines(self, linelist):
         self.write(data)
 
     def __getattr__(self, name):
+        if name == 'name':
+            return '{!r}'.format(self.buffer)
```

Comment:
This failed in py26 because the field index is mandatory... but I suggest just using `repr(self.buffer)` as it is shorter and simpler.